src/read.c

```diff
@@ -3623,12 +3626,14 @@ avifResult avifDecoderNextImage(avifDecoder * decoder)
             avifImageFreePlanes(decoder->image, AVIF_PLANES_A); // no alpha
         } else {
             if (decoder->data->alphaTileCount != 1) {
+                avifDiagnosticsPrintf(&decoder->diag, "decoder->data->alphaTileCount should be 1 but is %u", decoder->data->alphaTileCount);
                 return AVIF_RESULT_DECODE_ALPHA_FAILED;
             }
 
             avifImage * srcAlpha = decoder->data->tiles.tile[decoder->data->colorTileCount].image;
             if ((decoder->image->width != srcAlpha->width) || (decoder->image->height != srcAlpha->height) ||
                 (decoder->image->depth != srcAlpha->depth)) {
+                avifDiagnosticsPrintf(&decoder->diag, "decoder->image does not match srcAlpha in width, height, or bit depth");
```

Comment:
This one is interesting. If we plan to scale to `ispe`, this entire check might be spurious now, right?

Reply:
Hmm... Are you saying this error may not be a fatal error? One solution would be for the code that handles this error to clear the diagnostic message first.

Alternatively, we can document that decoder->diag.error should only be used after a public API function failure.

Reply:
I think it is safe to add this error while it continues to be a fatal error, but we should audit alpha size checks now that we are in this new world of ispe + ScalePlane.